File: services/schema/Distributed.Proposals.Schema/Proposal.cs
using System.ComponentModel.DataAnnotations.Schema;
using Distributed.Core.Schema;

namespace Distributed.Proposals.Schema;

[Table("Proposal")]
public class Proposal : Entity
{
    public int StatusId { get; set; }
    public int? PackageId { get; set; }
    public string Title { get; set; } = string.Empty;
}

File: apps/proposals/src/app/models/proposal.ts
import {
    FormBuilder,
    FormGroup,
    Validators
} from '@angular/forms';

import { Entity } from '@distributed/core';
import { GenerateEntityForm } from '@distributed/toolkit';


export interface Proposal extends Entity {
    statusId: number;
    packageId?: number;
    title: string;
}

export function GenerateProposalForm(proposal: Proposal, fb: FormBuilder): FormGroup {
    return fb.group({
        ...(GenerateEntityForm(proposal, fb)).controls,
        statusId: [proposal?.statusId ?? 0],
        packageId: [proposal?.packageId],
        title: [proposal?.title ?? '', Validators.required]
    })
}